zeppelin-zengine/src/main/java/org/apache/zeppelin/interpreter/ManagedInterpreterGroup.java

```diff
@@ -58,10 +58,19 @@ public InterpreterSetting getInterpreterSetting() {
   public synchronized RemoteInterpreterProcess getOrCreateInterpreterProcess(String userName,
                                                                              Properties properties)
       throws IOException {
+    if (remoteIntpProcessIsShutdown()) {
+      LOGGER.info("Check whether the InterpreterProcess has been shutdown.");
+      // clean invalid session and dirty data of interpreterSetting
+      close();
```

Comment:
same q about whether to close iff process is not running

Reply:
`remoteIntpProcessIsShutdown()==true`, Found that the remote interpreter process is unavailable, clean invalid session by close() function.